


/////////////////////////////////////////////////////////////背包模块//////////////////////////////////////////////////////////

message DBBagGrid
{
    optional uint32 ID = 1; // Item ID
    optional uint32 Num = 2; // Item Num
}

message DBBagGridList
{
    repeated DBBagGrid Grids = 1; // Grids
}

message DBBagInfo
{
    optional DBBagGridList GridList = 1; // CridList
}

/////////////////////////////////////////////////////////////背包模块///////////////////////////////////////////////////////////////


/////////////////////////////////////////////////////////////装饰背包模块//////////////////////////////////////////////////////////

message DBDecorateItem					//装饰结构
{
    optional uint32 id = 1; 			// 装饰 ID
    optional uint32 type = 2; 			// 装饰划分的类型
	optional uint64 EndTime = 3;		//装饰结束的时间
}

message DBDecorateItemList				//装饰结构数组
{
    repeated DBDecorateItem DecorateGrids = 1; // 装饰格子
}

message DBDecorateBagInfo				//装饰背包的总参数
{
    optional DBDecorateItemList BagGridList = 1; // 表示其中一个装饰背包,有该背包所有装饰商品组成
}

message	DBDecorateBagInfoList
{
	repeated DBDecorateBagInfo DecorateBagList = 1;		//装饰背包数组，因为装饰包里面有很多各式各样的装饰背包
}

message	DBDecorateBagModuleInfo							//装饰背包模块的总参数，所有类型的装饰背包集合在该结构中
{
	optional DBDecorateBagInfoList DecorateBagInfoList = 1;		//装饰背包数组
}

/////////////////////////////////////////////////////////////装饰背包模块///////////////////////////////////////////////////////////////



message DBRoleInfo											//人物角色大模块	
{
optional uint64 UID = 1; 									// UID
optional string Name = 2; 									// Name
optional uint32 Level = 3;									//游戏等级
optional uint32 Rank = 4;									//游戏段位等级
optional uint32 LevelExper = 5;								//游戏该等级的经验值
optional uint32 RankExper = 6;								//游戏该段位的经验
//如果还有别的模块就在这里填写他的结构

optional DBBagInfo BagInfo = 7;								//背包模块

optional DBDecorateBagModuleInfo DecorateBagModuleInfo = 8;	//装饰背包模块

}